src/tribler-common/tribler_common/utilities.py

```diff
@@ -27,6 +29,46 @@ def uri_to_path(uri):
 
 
 fts_query_re = re.compile(r'\w+', re.UNICODE)
+extract_tags_re = re.compile(r'\[\w+\]')
```

Comment:
**issue:** This regular expression does not allow special symbols encountered in pseudo-tags that we can find embedded in torrent titles. Most importantly, it does not allow dashes, like `tag-name`.

**suggestion:** at least we should allow dashes, but I suggest keeping tags outside the query string (and avoid parsing altogether), or at least use backward slashes (as I describe in another comment) and allow every symbol between backward slashed. We can validate tag names when a user adds a new tag. It is not necessary to duplicate checks when we perform a search.

Reply:
Fixed by the following regex: 
```python
tags_re = re.compile(r'^\s*(?:\[[^\s\[\]]{3,50}\]\s*)+')
```